nodejs/koa/app/app.js

```diff
@@ -11,9 +11,7 @@ console.log("Starting KOA app");
 app.on("error", (error) => {
   appsignal
     .tracer()
-    .currentSpan()
-    .addError(error)
-    .close()
+    .setError(error)
```

Comment:
This may need to be replaced with sendError I think? I'll have to test.

Reply:
I was confused by the use of `close` in the old code. I don't think that was necessary, because koa will create the span and close it.

Testing this I get no data in AppSignal though. The rootspan is always a `NoopSpan`. Are you seeing the same thing?

Reply:
No, I'm able to see the performance traces. Maybe the agent didn't startup properly in your container?

Reply:
You’ll need to use `setError` here, as you’re adding the error to the existing span. The span should then be closed, as we’ve run into an error and can’t continue adding more performance measurements to this span (like we do in the [Plug instrumentation](https://github.com/appsignal/appsignal-elixir-plug/blob/main/lib/appsignal_plug.ex#L59)). I think you still need the call to `close`, there.

Reply:
The problem was that the rootSpan was being lost during the context change. Already fixed on https://github.com/appsignal/appsignal-nodejs/pull/446 latest update.

Reply:
In request time everything goes perfect but when a context switch happens, in this case, with the .on("error"... thingie, the execution ID changes and a new ScopeManager instance is created. I updated the constructor to setup the root when you come from a different context as it already had the same for the other spans.